src/Widgets/PreferencesDialog.vala

```diff
@@ -180,6 +191,11 @@ public class ENotes.PreferencesDialog : Gtk.Dialog {
             return false;
         });
 
+        keep_sidebar_switch.state_set.connect ((state) => {
+            settings.keep_sidebar_visible = state;
```

Comment:
Could you toggle the sidebar here IF the user is in Editing Mode? That way, if you're changing the setting while you're in Edit Mode, it will change automatically :) 

You can get the current mode by: `ENotes.ViewEditStack.current_mode == ENotes.Mode.EDIT`